clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1231,10 +1241,13 @@ public void assign(Collection<TopicPartition> partitions) {
                         client.pollNoWakeup();
                     }
 
+                    this.kafkaConsumerMetrics.recordPollEnd(time.milliseconds());
                     return this.interceptors.onConsume(new ConsumerRecords<>(records));
                 }
             } while (timer.notExpired());
 
+            this.kafkaConsumerMetrics.recordPollEnd(time.milliseconds());
```

Comment:
Wouldn't it make more sense to do this in the `finally`?

Reply:
Good catch. I moved the `recordPollStart` call to the beginning of the try block also.